gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/organization/UserResource.java

```diff
@@ -120,31 +124,30 @@ public Response deleteUser() {
     @ApiResponse(responseCode = "404", description = "User not found")
     @ApiResponse(responseCode = "500", description = "Internal server error")
     @Permissions(@Permission(value = RolePermission.ORGANIZATION_USERS, acls = RolePermissionAction.READ))
-    public List<UserGroupEntity> getUserGroups() {
+    public List<UserGroupEntity> getUserGroups(@QueryParam("q") String query) {
         // Check that user belongs to current organization
         userService.findById(GraviteeContext.getExecutionContext(), userId);
+        LOGGER.debug("Search groups for user: {} with query: {}", userId, query);
+        Set<GroupEntity> groups = groupService.findByUserAndEnvironment(userId, query);
+        List<UserGroupEntity> result = new ArrayList<>();
+        groups.forEach(groupEntity -> {
+            UserGroupEntity userGroupEntity = new UserGroupEntity();
```

Comment:
I would also add (if it's possible) the environmentId in the `UserGroupEntity` response.